test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -61,19 +63,36 @@ public static void main(String[] args) throws Exception {
 
         try {
             // create test directory
-            String tmpDir = System.getProperty("user.home");
-            testDir = new File[1];
-            testDir[0] = new File(tmpDir, "testDir");
-            if (!testDir[0].exists())
-                testDir[0].mkdir();
-            testDir[0].deleteOnExit();
-
-            // create temporary files inside testDir
-            tempFile = new File[5];
+            String tmpDir = System.getProperty("java.io.tmpdir");
```

Comment:
This is the purpose of the [*scratch* directory in jtreg](https://openjdk.org/jtreg/faq.html#scratch-directory), and it's automatically [clean up after the test](https://openjdk.org/jtreg/writetests.html#cleanFiles): <q cite="https://openjdk.org/jtreg/writetests.html#cleanFiles">jtreg will automatically clean up any files written in the scratch directory.</q>

The test is designed for jtreg, so the simplest solution is *to use the current directory*.

If some one runs the test directly, without using jtreg, then it's their task to set up the current directory or clean up after the test.

Reply:
For some reasons when I used the current directory i.e. scratch, test failed in my local machine.

Reply:
> For some reasons when I used the `current directory i.e. scratch`, test failed in my local machine. `String tmpDir = System.getProperty(".");`

How does it fail?  
Is it because `System.getProperty(".")` returns `null`? The property `"."` doesn't exist, does it?

Use `new File(".")` which resolves to the current directory. Alternatively, remove references to paths, just create directories and files, they will be created in the current directory if you don't specify an absolute path.

Reply:
> How does it fail? Is it because `System.getProperty(".")` returns `null`? The property `"."` doesn't exist, does it?

No, temp files and directories are created in `scratch` directory. But the files gets deselected after selection which returns as no file selected and test failed.

Reply:
Surprise to me also. It failed every single time when temporary files are created in `scratch`. 

>The test isn't as stable as it should be.

Unable to trace what is missing in the test.

Reply:
It took a long while for me to figure it out. What matters is whether the `File` object is created with an absolute path or not. By using `System.getProperty("java.io.tmpdir")` as the base path, you create an absolute path for the parent directory.

Yet if you use pass `null` as the first parameter (as I said, there's no system property named ".", so `tmpDir` is `null`) or use the constructor with one parameter `new File("testDir")`, the file has relative path. This causes some issues…

When you create `JFileChooser` object, you *always* pass `testDir` as its starting directory. Then, for each case, you set it once again. Here's what I see in a `PropertyChangeListener`:

```text
>> setCurrentDirectory
directoryChanged: /home/aivanov/jdk-dev/testDir/testDir -> testDir
<< setCurrentDirectory

SelectedFileChangedProperty: /home/aivanov/jdk-dev/testDir/subDir_3 -> null
SelectedFilesChangedProperty: [Ljava.io.File;@3fed803b -> null
directoryChanged: testDir -> /home/aivanov/jdk-dev/testDir
SelectedFileChangedProperty: null -> null
SelectedFilesChangedProperty: null -> [Ljava.io.File;@1bf88ed6
```

This is why see a directory gets selected but then becomes unselected again.

Getting the absolute file — `testDir = new File("testDir").getAbsoluteFile()` — resolves this problem, and the test starts to work in the current directory. I tested it by running with jtreg, then the current directory is the `scratch` directory, and as a stand-alone app, in this case the current directory is whatever the current directory is.

So either way works as long as the starting `File` object is created with _an absolute path_.